### ISSUE TITLE
search & replace with large files freezes

### ISSUE THREAD

**Issue by larsVolta** [notepad-plus-plus/notepad-plus-plus#10260]
### Description of the Issue

I updated npp from 7.9.5 to 8.1.2

When using search & replace for large files (5 million lines, 11 characters each line - replace 2 characters on each line) I’m waiting 15-20 minutes without progress and have to stop npp hard.

I now downgraded to 7.9.5 again (from which I updated) and the same procedure takes about 3-4 minutes.

### Steps to Reproduce the Issue
txt-file, 5 million lines, each line looks similar to this (random letters): ABC;DEF;GHI. Now I want to replace “;” with “” (empty String)

My settings (for 8.1.2 and 7.9.5):

https://community.notepad-plus-plus.org/assets/uploads/files/1627303378851-f734b08c-03eb-4aa4-b6ae-ca6e7740bfc3-grafik.png

### Update Debug-Info:

Notepad++ v8.1.2   (64-bit)
Build time : Jul 16 2021 - 01:59:50
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19042.1110
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll 

JSON Viewer 1.34, Mime tools 2.6, Npp Converter 4.3, NppExport 0.3

**Comment by donho** [comment]
@larsVolta 
Could you remove all the plugins then try again?

**Comment by DavePCoder** [comment]
I have the same issue - doing a replace all in large file (mine is 15Mb XML) results in notepad++ appearing to hang.  Regardless of whether the speed can be improved, it would be good to have a progress bar and found count, so you can tell that it is still working.  Note also in a file this big, when typing any characters the suggestions box takes ages to come up and delays the typing continuing.

**Comment by larsVolta** [comment]
Vielen Dank für Ihre E-Mail,


vom 03.08.2021 - 20.08.2021 bin ich nicht im Büro und kann in dieser Zeit Ihre E-Mail leider nicht beantworten. Leider können wir in der Programmierung in dieser Zeit auch keine neuen Projekte bearbeiten. 


Bitte wenden Sie sich in dringenden Fällen an meine Kollegin Lydia Weniger ***@***.*** - Telefon: +49 3722 598 75 22). 

Mit freundlichen Grüßen
Finzel & Schuck GmbH

i.A. Lars Hamann
Lead Developer 



Finzel & Schuck GmbH I Paul-Seydel-Str. 13 I 09212 Limbach-Oberfrohna
Tel  +49 3722 59 875 21 I Fax +49 3722 6002 79
***@***.*** I www.finzelundschuck.de
 
Geschäftsführer: Alexander Schuck, Michael Walther
Registergericht: AG Chemnitz HRA 27353
USt.-ID: DE282913803 I Steuernr.: 227/108/06245

**Comment by larsVolta** [comment]
> @larsVolta
> Could you remove all the plugins then try again?

I‘ll try it when I‘m back from vacation

**Comment by donho** [comment]
@larsVolta 

Could you try [fixed x32](http://download.notepad-plus-plus.org/repository/MISC/npp_fix_perf_32.zip) or [x64  binary](http://download.notepad-plus-plus.org/repository/MISC/npp_fix_perf_64.zip) ?
I believe this issue is fixed in the binaries I provide above.

**Comment by felix-22** [comment]
Yeah, it's fixed - with "Steps to Reproduce the Issue" provided by the OP:
```

ver   | N++       |  N++
      | running   |  not running
      | as admin  |  as admin

64bit     53 s        93 s
32bit    105 s       144 s
```

By the way it seems:
- the difference between when N++ is running as admin vs when N++ is not running as admin grows with the file size and numer of replace operations per line, the difference has growed to 30%-44% as compared to 17%-23% from the previous case*
- the same happens with the difference between 64bit and 32bit, it has growed to 35%-50% as compared to 14%-21% from the previous case*

*the previous case here: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10398#issuecomment-901666287 + https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10398#issuecomment-901818774

**Comment by Nealix58** [comment]
I hit this exact same issue in Version 8.6.7, Aug 1, 2024.   Rolling back to version 7.9.5 fixed the issue for me.
So the bug seems to have re-surfaced again.

Details:  I had a 5 megabyte text file, an XML log for radio contacts, where I wanted search and replace to remove
the CR LF from the end of each line.    After clicking "Replace All", the task did not complete even after 12 minutes.
Opening Task Manager on Windows 11 x64, showed approx 7% CPU utilization ongoing for Notepad++, and the
task never completed.  I killed notepad++ in task manager.

I uninstalled Version 8.6.7 and rolled down to version 7.9.5.
Rerunning the attempt to search and replace \r\n with "nul" worked fine and completed in less than 1.5 minutes.

Neal
